sqlglot/dialects/tsql.py

```diff
@@ -326,24 +326,21 @@ def _parse(args: t.List[exp.Expression]) -> exp.Expression:
 
 # https://learn.microsoft.com/en-us/sql/t-sql/functions/parsename-transact-sql?view=sql-server-ver16
 def _build_parsename(args: t.List) -> exp.SplitPart | exp.Anonymous:
-    anonymous = exp.Anonymous(this="PARSENAME", expressions=args)
-    # Not correct number of arguments or
-    # any argument is not literal
-    if len(args) != 2 or isinstance(args[0], exp.Column) or isinstance(args[1], exp.Column):
-        return anonymous
-    arg_this: exp.Expression = args[0]
-    text = arg_this.name
-    part_count = len(text.split("."))
-    if part_count > 4:
-        return anonymous
-
-    arg_partnum: exp.Expression = args[1]
-    part_num = int(arg_partnum.name)
-    length = 1 if isinstance(arg_this, exp.Null) else part_count + 1  # Reverse index
-    idx = 0 if isinstance(arg_this, exp.Null) else part_num
-    return exp.SplitPart(
-        this=arg_this, delimiter=exp.Literal.string("."), part_num=exp.Literal.number(length - idx)
-    )
+    # PARSENAME(...) will be stored into exp.SplitPart if:
+    # - All args are literals
+    # - The part index (2nd arg) is <= 4 (max valid value, otherwise TSQL returns NULL)
+    if len(args) == 2 and all(isinstance(arg, exp.Literal) for arg in args):
```

Comment:
Can also do `all(arg.is_string for arg in args)` here.

Reply:
We don't want them all to be strings, the 2nd arg of `PARSENAME` (and 3rd arg of `SPLIT_PART`) is an integer

Reply:
Ah I see, my bad.